themes/docs-new/layouts/shortcodes/unified_mode.md

```diff
@@ -0,0 +1,192 @@
+sr = edit_resource(:dhcp_subnet, "#{new_resource.name}_sharedsubnet_#{subnet}") do
+  owner new_resource.owner
+  group new_resource.group
+  mode new_resource.mode
+
+  ip_version new_resource.ip_version
+  conf_dir new_resource.conf_dir
+  shared_network true
+end
+
+properties.each do |property, value|
+  sr.send(property, value)
+end
+```
+
+To fix resources that change sub-resources during their declaration, you can:
+
+- Apply properties in the codeblock (preferred)
+- Run the resource explicitly (not preferred)
+
+#### Apply Properties in the Codeblock
+
+This pattern declares the sub-resource in one codeblock and then changes it in the next codeblock. This is the preferred pattern in unified mode because all resources execute in order at compile time.
+
+```ruby
+dhcp_subnet "#{new_resource.name}_sharedsubnet_#{subnet}" do
+  owner new_resource.owner
+  group new_resource.group
+  mode new_resource.mode
+
+  ip_version new_resource.ip_version
+  conf_dir new_resource.conf_dir
+  shared_network true
+
+  properties.each do |property, value|
+    send(property, value)
+  end
+end
+```
+
+#### Run the Resource Explicitly
+
+Another solution is to continue saving the resource as a variable, declaring `action :nothing` within the codeblock and then explicitly running the action in another codeblock.
+
+The pattern of saving a resource as a variable and then forcing it to run at compile time with an explicit `run_action` works as it has in the past, but it not a preferred pattern. Unified mode forces resource execution to compile time by default, which makes this pattern redundant.
+
+```ruby
+sr = edit_resource(:dhcp_subnet, "#{new_resource.name}_sharedsubnet_#{subnet}") do
+  owner new_resource.owner
+  group new_resource.group
+  mode new_resource.mode
+
+  ip_version new_resource.ip_version
+  conf_dir new_resource.conf_dir
+  shared_network true
+
+  action :nothing
+end
+
+properties.each do |property, value|
+  sr.send(property, value)
+end
+
+# Run the action explicitly
+sr.run_action(:create)
+```
+
+## Actions on Later Resources
+
+Since unified mode executes your resource as it is compiled, you cannot send immediate notifications to resources declared later.
+
+### `:immediate` Notifications to Later Resources
+
+Unified mode delays immediate notifications to later resources.
+In unified mode the Chef Infra Client saves immediate notifications and executes them when the later resource is parsed. Immediate notifications to prior resources and delayed notifications behave the same as they did before unified mode.
+
+Chaining immediate notifications results in the same behavior as before unified mode. Changing the order of resources results does not change the results of a notification chain. Chained notifications in the same order during the Chef Infra Client run and immediately sends a notification to a target resource that was parsed earlier and saves a notification and sends it immediately after it parses a later resource.
```

Comment:
I think the meaning of the second sentence got a bit mangled somewhere and the whole thing is a bit awkward to read.

```suggestion
Chaining immediate notifications results in the same behavior as before unified mode.  While the resources are not immediately notified, instead as they are parsed the notifications fire sequentially and the result is the same.  If the parse order and the intended execution order are very different the results may be different and reflect parse order.
```

My second sentence clearly needs work there.